src/app/shared/services/ministry-admin/ministry-admin.service.spec.ts

```diff
@@ -1,19 +1,144 @@
-import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
 import { TestBed } from '@angular/core/testing';
 import { NgxsModule } from '@ngxs/store';
+
+import { AdminRoles } from 'shared/enum/admins';
+import { BaseAdminParameters } from 'shared/models/admin.model';
+import { MinistryAdmin } from 'shared/models/ministry-admin.model';
 import { MinistryAdminService } from './ministry-admin.service';
 
 describe('MinistryAdminService', () => {
+  const baseApiUrl = `/api/v1/${AdminRoles.ministryAdmin}`;
   let service: MinistryAdminService;
+  let httpTestingController: HttpTestingController;
+  const mockMinistryAdmin: MinistryAdmin = {
+    id: 'id',
+    email: 'email',
+    firstName: 'firstName',
+    lastName: 'lastName',
+    institutionId: 'institutionId',
+    phoneNumber: 'phoneNumber'
+  };
 
   beforeEach(() => {
     TestBed.configureTestingModule({
       imports: [HttpClientTestingModule, NgxsModule.forRoot([])]
     });
     service = TestBed.inject(MinistryAdminService);
+    httpTestingController = TestBed.inject(HttpTestingController);
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should get admin profile', () => {
+    service.getAdminProfile().subscribe((profile) => {
+      expect(profile).toEqual(mockMinistryAdmin);
+    });
+
+    const req = httpTestingController.expectOne(`${baseApiUrl}/Profile`);
+    req.flush(mockMinistryAdmin);
+
+    expect(req.request.method).toEqual('GET');
+  });
+
+  it('should get admin by id', () => {
+    const adminId = 'id';
+
+    service.getAdminById(adminId).subscribe((ministryAdmin) => {
+      expect(ministryAdmin).toEqual(mockMinistryAdmin);
```

Comment:
use DoneFn after asynchronous checks
and in other places too

Reply:
> use DoneFn after asynchronous checks and in other places too

Isn't `DoneFn` appear in Jasmine, not Jest? Also my editor can't find the interface

Reply:
It's actually the same thing,
https://jestjs.io/docs/asynchronous#callbacks